src/ui/control/geolocate_control.js

```diff
@@ -251,9 +263,33 @@ class GeolocateControl extends Evented {
 
     _updateMarker(position: ?Position) {
         if (position) {
-            this._userLocationDotMarker.setLngLat([position.coords.longitude, position.coords.latitude]).addTo(this._map);
+            const center = new LngLat(position.coords.longitude, position.coords.latitude);
+            this._accuracyCircleMarker.setLngLat(center).addTo(this._map);
+            this._userLocationDotMarker.setLngLat(center).addTo(this._map);
+            this._accuracy = position.coords.accuracy;
+            if (this.options.trackUserLocation && this.options.showAccuracyCircle) {
```

Comment:
We don't need to check for `trackUserLocation` because you can still have an accuracy circle and user location dot shown with `trackUserLocation=false` and in that case we still need to update the radius after we have received the geolocation position and set `this._accuracy`.

Reply:
In fact it wouldn't hurt to try and add a unit test to catch this if possible.